application/F3DOptionsParser.cxx

```diff
@@ -274,12 +255,60 @@ void ConfigurationOptions::PrintPluginsScan()
 
 //----------------------------------------------------------------------------
 void ConfigurationOptions::GetOptions(F3DAppOptions& appOptions, f3d::options& options,
-  std::vector<std::string>& inputs, std::string filePathForConfigBlock, bool allOptionsInitialized,
-  bool parseCommandLine)
+  std::vector<std::string>& inputs, const std::string& filePathForConfigBlock,
+  bool allOptionsInitialized, bool parseCommandLine)
 {
   inputs.clear(); /* needed because this function is called multiple times */
 
-  this->FilePathForConfigBlock = std::move(filePathForConfigBlock);
+  /* start with an empty config ... */
+  ConfigDict tmpConfig;
+
+  const auto update = [&](const ConfigDict& config)
+  {
+    for (const auto& [key, value] : config)
+    {
+      const std::string del = tmpConfig.count(key) ? key + ": " + tmpConfig[key] : "";
+      const std::string add = std::string(key).append(": ").append(value);
+      if (add == del)
+      {
+        f3d::log::debug("= ", add);
+      }
+      else
+      {
+        if (!del.empty())
+        {
+          f3d::log::debug("- ", del);
+        }
+        f3d::log::debug("+ ", add);
+      }
+
+      tmpConfig[key] = value;
+    }
+  };
+
+  /* apply global entries */
+  for (const auto& [conf, source, pattern] : this->GlobalConfigEntries)
```

Comment:
`pattern` here is always `global` right ?

Not saying we should change stuff, just checking I'm following.

Reply:
yes, I kept the same tuple typing with the pattern but for the global blocks it's basically unused, I'll add it to the comment